plugins/steelseries/fu-steelseries-fizz-tunnel.c

```diff
@@ -0,0 +1,395 @@
+{
+	FuDevice *parent = fu_device_get_parent(device);
+	guint8 fs = STEELSERIES_FIZZ_FILESYSTEM_MOUSE;
+	guint8 id = STEELSERIES_FIZZ_MOUSE_FILESYSTEM_BACKUP_APP_ID;
+	gsize bufsz = 0x27000;
+	g_autoptr(FuFirmware) firmware = fu_steelseries_firmware_new();
+	g_autoptr(GBytes) blob = NULL;
+	g_autofree guint8 *buf = NULL;
+
+	fu_progress_set_id(progress, G_STRLOC);
+	fu_progress_add_step(progress, FWUPD_STATUS_DEVICE_READ, 100);
+
+	buf = g_malloc0(bufsz);
+	if (!fu_steelseries_fizz_read_access_file(parent,
+						  TRUE,
+						  fs,
+						  id,
+						  buf,
+						  bufsz,
+						  fu_progress_get_child(progress),
+						  error))
+		return NULL;
+	fu_progress_step_done(progress);
+
+	if (g_getenv("FWUPD_STEELSERIES_FIZZ_VERBOSE") != NULL)
+		fu_common_dump_raw(G_LOG_DOMAIN, "Firmware", buf, bufsz);
+	blob = g_bytes_new_take(g_steal_pointer(&buf), bufsz);
+	if (!fu_firmware_parse(firmware, blob, FWUPD_INSTALL_FLAG_NONE, error))
+		return NULL;
+
+	/* success */
+	return g_steal_pointer(&firmware);
+}
+
+static void
+fu_steelseries_fizz_tunnel_set_progress(FuDevice *self, FuProgress *progress)
+{
+	fu_progress_set_id(progress, G_STRLOC);
+	fu_progress_add_step(progress, FWUPD_STATUS_DEVICE_RESTART, 0); /* detach */
+	fu_progress_add_step(progress, FWUPD_STATUS_DEVICE_WRITE, 82);	/* write */
+	fu_progress_add_step(progress, FWUPD_STATUS_DEVICE_RESTART, 0); /* attach */
+	fu_progress_add_step(progress, FWUPD_STATUS_DEVICE_BUSY, 18);	/* reload */
+}
+
+static void
+fu_steelseries_fizz_tunnel_class_init(FuSteelseriesFizzTunnelClass *klass)
+{
+	FuDeviceClass *klass_device = FU_DEVICE_CLASS(klass);
+
+	klass_device->attach = fu_steelseries_fizz_tunnel_attach;
+	klass_device->probe = fu_steelseries_fizz_tunnel_probe;
+	klass_device->setup = fu_steelseries_fizz_tunnel_setup;
+	klass_device->poll = fu_steelseries_fizz_tunnel_poll;
+	klass_device->write_firmware = fu_steelseries_fizz_tunnel_write_firmware;
+	klass_device->read_firmware = fu_steelseries_fizz_tunnel_read_firmware;
+	klass_device->set_progress = fu_steelseries_fizz_tunnel_set_progress;
+}
+
+static void
+fu_steelseries_fizz_tunnel_init(FuSteelseriesFizzTunnel *self)
+{
+	fu_device_set_version_format(FU_DEVICE(self), FWUPD_VERSION_FORMAT_TRIPLET);
+	fu_device_add_flag(FU_DEVICE(self), FWUPD_DEVICE_FLAG_UPDATABLE);
+	fu_device_add_flag(FU_DEVICE(self), FWUPD_DEVICE_FLAG_CAN_VERIFY_IMAGE);
+	fu_device_add_internal_flag(FU_DEVICE(self), FU_DEVICE_INTERNAL_FLAG_USE_PARENT_FOR_OPEN);
+	fu_device_add_internal_flag(FU_DEVICE(self), FU_DEVICE_INTERNAL_AUTO_PAUSE_POLLING);
+	fu_device_add_protocol(FU_DEVICE(self), "com.steelseries.fizz");
+	fu_device_set_logical_id(FU_DEVICE(self), "fizz-tunnel");
+	fu_device_set_install_duration(FU_DEVICE(self), 38);				  /* 38 s */
+	fu_device_set_remove_delay(FU_DEVICE(self), FU_DEVICE_REMOVE_DELAY_RE_ENUMERATE); /* 10 s */
+	fu_device_set_poll_interval(FU_DEVICE(self), 10000); /* 1 min */
```

Comment:
that's 10 seconds, no?

Reply:
oops, that was for testing the flag `FU_DEVICE_INTERNAL_AUTO_PAUSE_POLLING`...